components/telemetry-operator/controllers/logpipeline_controller.go

```diff
@@ -311,18 +308,27 @@ func (r *LoggingConfigurationReconciler) deleteFluentBitPods(ctx context.Context
 	return nil
 }
 
-// Merge list of Sections to single FluentBit configuration.
-func generateFluentBitConfig(sections []telemetryv1alpha1.Section) string {
-	var result string
-	for _, section := range sections {
-		result += section.Content + "\n\n"
+// Merge FluentBit parsers, filters and outputs to single FluentBit configuration.
+func mergeFluentBitConfig(logPipeline *telemetryv1alpha1.LogPipeline) string {
+	var sb strings.Builder
+	for _, parser := range logPipeline.Spec.Parsers {
```

Comment:
Having (multiline) parsers in the same configuration as filters and outputs would not work. Have a look at the configuration file [docs](https://docs.fluentbit.io/manual/administration/configuring-fluent-bit/configuration-file). The parsers file (which Iirc contains parsers and multi-line parsers), is referenced in the service configuration. So we should write this to a different CM that is consumed by the DaemonSet.

Reply:
Okay, will adjust that :) Is a different CM necessary? Couldn't we write it into the existing CM with a new key (e.g. `parsers.conf`)?

Reply:
That would make things a bit more complicated since different LogPipeline CRs are already rendered to keys in one single CM to be consumed by Fluent Bit through an `@INCLUDE` statement. Having the parsers file in the same CM would require some rather dirty hacks to figure out what is a "normal" configuration snippet and what the parsers file.